pulsar/src/utils.rs

```diff
@@ -388,3 +388,37 @@ impl<'de> Deserialize<'de> for Rewards {
         Ok(Rewards(value))
     }
 }
+
+/// move data from old directory (if any) to new directory, or else just create
+/// the new directory.
+pub(crate) fn create_or_move_data(old_dir: PathBuf, new_dir: PathBuf) -> Result<()> {
```

Comment:
Please write test for this method since if this method does not work properly node data could be corrupted or incomplete.

Reply:
Added tests [here](https://github.com/subspace/pulsar/commit/9b2539e06ed38b12251ef7de1c25564b0090af32).